### PR TITLE
Some bikeshed cleanup, remove references to unrelated specs

### PR DESCRIPTION
**Pull request by mpb**
Right now, `<a>root<a/>` and `<a>target</a>` go to 
```
<a href="https://dom.spec.whatwg.org/#concept-tree-root">root</a>
<a href="https://dom.spec.whatwg.org/#concept-pi-target">target</a>
```
They should be local concepts within Intersection Observer.

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -131,28 +131,25 @@ The IntersectionObserverCallback</h3>
...
-{{IntersectionObserverInit/rootMargin}}
-or {{threshold}} values
-for each <a>target</a>
-seems to introduce more complexity without solving additional use-cases.
-Per-{{observe()}} options could be provided in the future if V2 introduces a need for it.
+The {{IntersectionObserver}} interface can be used to observe changes in the
+intersection of a <a>target</a> {{Element}} and a {{IntersectionObserver/root}}
+{{Element}} (or the top-level document's viewport).
+
+Note: In {{MutationObserver}}, the {{MutationObserverInit}} options are passed
```
Comment (szager-chromium) [inline index.bs]: This section should be updated based on the comments in issue #81, but that can go in a different patch.

Comment (mpb) [inline index.bs]: Agreed, but there still doesn't seem to be consensus on that thread?

Comment (szager-chromium) [inline index.bs]: I think the consensus is:

For typical usage, developers should feel free to create one observer per target.

In the rare case where many targets are observing the same root with the same parameters; and there is a likelihood that multiple notifications will be generated during a single run of the algorithm; then the grouping of notifications in the callback may offer a performance advantage.

Comment (mpb) [inline index.bs]: Ok, but none of that changes the API as spec'ed, right?

Comment (szager-chromium) [inline index.bs]: Correct; it's just to anticipate a likely faq and source of developer confusion.
